modules/home-environment.nix

```diff
@@ -374,7 +380,11 @@ in
             export __HM_SESS_VARS_SOURCED=1
 
             ${config.lib.shell.exportAll cfg.sessionVariables}
-          '' + cfg.sessionVariablesExtra;
+          ''
+          + (if cfg.sessionPath != "" then ''
```

Comment:
Is it possible to pass a list of paths as well, and you concatenate the list with `:`?

Reply:
I'm not sure of Nix's union-type capabilities, but the [envVar](https://github.com/NixOS/nixpkgs/blob/master/lib/types.nix#L221) type is simply a string. The only benefit is that the paths are concatenated together with a `:` on merging of the values. If you wish to add many paths in a single call, your best bet is to do something like

```nix
home.sessionPath = "foo/bar:foo/baz";
```

A `listOf` approach is certainly doable, as those are merged together as well, but I'm not sure which is preferential. We'd have to do the concatenation manually instead of relying on the type to do it for us.

Reply:
`lib.strings` already provides `makeBinPath` to construct a `$PATH`

Reply:
`makeBinPath` appends `/bin` to the given values, which isn't necessarily what a user might want.

I've changed it to a list that uses `concatStringsSep` to allow using a `listOf str` instead.